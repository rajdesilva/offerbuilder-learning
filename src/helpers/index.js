export * from './api-config';
export * from './history';
