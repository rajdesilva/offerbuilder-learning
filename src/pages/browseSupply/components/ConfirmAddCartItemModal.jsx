import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  browseSupplyActions,
  propertyCartActions,
  supplySearchActions,
} from "../actions";
import Modal from "antd/lib/modal/Modal";
import { InfoCircleOutlined } from "@ant-design/icons";
import { FormattedMessage } from "react-intl";
import { store } from "../../../redux/store";
import { isEqual } from "lodash";
import { newOfferActions } from "../../offers/actions";
import styles from "./css/ConfirmAddCartItemModal.module.less";

export function ConfirmAddCartItemModal({ row, hideModal, isEditFlow }) {
  const dispatch = useDispatch();
  const [visible, setVisible] = useState(true);
  const currencyCode = useSelector(
    (state) => window.getValue(state, "searchparams.currencyCode"),
    isEqual
  );
  const destination = useSelector(
    (state) => window.getValue(state, "searchparams.destination"),
    isEqual
  );
  const lastSearchedDistance = useSelector(
    (state) => window.getValue(state, "searchparams.distance"),
    isEqual
  );
  const handleOk = () => {
    dispatch({
      type: newOfferActions.NEW_OFFER_DEEPLINK_DESTINATION,
      payload: destination,
    });
    dispatch({
      type: supplySearchActions.SET_SELECTED_CURRENCY,
      payload: currencyCode,
    });
    dispatch({
      type: newOfferActions.NEW_OFFER_LAST_SEARCH_DISTANCE,
      payload: lastSearchedDistance,
    });
    dispatch({
      type: browseSupplyActions.ADD_TEMP_EDITED_BRANDS,
      payload: {
        key: row ? row.propertyCode + row.supplier + row.channel : "",
        brands: store.getState().searchparams.brands,
      },
    });
    dispatch({
      type: propertyCartActions.ADD_PROPERTY_TO_CART,
      payload: row,
    });
    hideModal();
    setVisible(false);
  };

  const cancelModal = () => {
    hideModal();
    setVisible(false);
  };

  return (
    <Modal
      title={
        <div>
          <InfoCircleOutlined className={styles.cartPopupInfoIcon} />
          <FormattedMessage id="nemo.property.was.already.added" />
        </div>
      }
      visible={visible}
      destroyOnClose
      data-testid="property-already-added-modal"
      onOk={handleOk}
      onCancel={cancelModal}
    >
      <p>
        <FormattedMessage id="nemo.already.added.msg" />
      </p>
    </Modal>
  );
}
