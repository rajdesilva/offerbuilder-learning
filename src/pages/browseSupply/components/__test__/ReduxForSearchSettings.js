export const ReduxForSearchSettings = {
  searchparams: {
    destination: {},
    los: 1,
    distance: 10,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [{}],
      },
    ],
    dateRange: {
      startDate: "05/17/2021",
      endDate: "07/17/2021",
    },

    lcn: false,
    remainingCapitalPool: "",
    sortBy: {
      sortOrder: "ASCENDING",
      sortCriteria: "NAME",
    },
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "PCT",
        name: "PCT",
      },
      {
        id: "PCT",
        name: "ATG",
      },
      {
        id: "DEMO_OFFERBUILDER",
        name: "DEMO_OFFERBUILDER",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
