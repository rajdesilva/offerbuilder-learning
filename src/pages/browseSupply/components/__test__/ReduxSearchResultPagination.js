export const PaginationStore = {
  searchedproperties: {
    loading: false,
    properties: [{}],
    totalMatch: 15,
  },
  searchparams: {
    pageOffset: 0,
    pageSize: 10,
  },
};
