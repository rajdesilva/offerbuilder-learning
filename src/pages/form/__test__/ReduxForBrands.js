export const ReduxForBrands = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: false,
    target: {
      suppliers: [],
      channels: [],
    },
    brands: [
      {
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
        id: "idb",
        name: "Internal Demo Brand",
      },
    ],
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
  sourceinfo: {
    loading: false,
    brands: [
      {
        id: "idb",
        name: "Internal Demo Brand",
        storefronts: [
          {
            id: "ids",
            name: "Internal Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
      {
        id: "cdb",
        name: "Client Demo Brand",
        storefronts: [
          {
            id: "cds",
            name: "Client Demo Storefront",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids2",
            name: "Best Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
          {
            id: "ids3",
            name: "PC Travel",
            suppliers: [
              {
                id: "ntp",
                name: "NTP",
                channels: [
                  {
                    id: "DEMO_OFFERBUILDER",
                    name: "DEMO_OFFERBUILDER",
                  },
                  {
                    id: "DEMO_INDIA_CH",
                    name: "DEMO_INDIA_CH",
                  },
                  {
                    id: "NEMO_CH_01",
                    name: "NEMO_CH_01",
                  },
                  {
                    id: "NEMO_CH_02",
                    name: "NEMO_CH_02",
                  },
                  {
                    id: "NEMO_CH_03",
                    name: "NEMO_CH_03",
                  },
                  {
                    id: "NEMO_CH_04",
                    name: "NEMO_CH_04",
                  },
                  {
                    id: "NEMO_CH_05",
                    name: "NEMO_CH_05",
                  },
                  {
                    id: "NEMO_CH_06",
                    name: "NEMO_CH_06",
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
};
