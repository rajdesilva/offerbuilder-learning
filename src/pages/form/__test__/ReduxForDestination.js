export const reduxStore = {
  supplysearchParams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [],
      channels: [],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channels: {
    // loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: null,
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "AT_KINDERHOTELS",
        name: "Ihr Bett im Allg√§u",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
