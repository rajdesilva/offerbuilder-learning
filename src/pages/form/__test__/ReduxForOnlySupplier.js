export const ReduxForOnlySupplier = {
  searchparams: {
    destination: {},
    los: 1,
    onlySupplier: true,
    target: {
      suppliers: [
        {
          id: "ntp",
          name: "NTP",
        },
      ],
      channels: [
        {
          id: "AT_CORDIAL",
          name: "",
        },
      ],
    },
    dateRange: {
      startDate: "2020-09-18",
      endDate: "2020-11-24",
    },

    lcn: false,
    remainingCapitalPool: "",
  },
  channelinfo: {
    loading: false,
    channels: [
      {
        id: "AT_CORDIAL",
        name: "",
        type: "Portal",
        status: "HIDDEN",
      },
      {
        id: "india_channel",
        name: "indian channel",
        type: "Portal",
        status: "HIDDEN",
      },
    ],
  },
};
