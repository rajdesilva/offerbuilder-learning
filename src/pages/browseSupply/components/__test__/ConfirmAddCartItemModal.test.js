import React from "react";
import { createStore } from "redux";
import { render, fireEvent, act } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import { ConfirmAddCartItemModal } from "../ConfirmAddCartItemModal";
import { tempStore } from "../../../offers/components/__test__/tempStore";

describe("ConfirmAddCartItemModal component test cases", () => {
  test("create ConfirmAddCartItemModal components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    const wrapper = render(
      <MemoryRouter>
        <ConfirmAddCartItemModal />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
  });
  test("ConfirmAddCartItemModal - verify component modal hideModal functionality", async () => {
    const hideModal = jest.fn();

    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const wrapper = render(
      <MemoryRouter initialEntries={["/browse-supply/search-results"]}>
        <ConfirmAddCartItemModal
          {...{
            hideModal,
            row: {},
          }}
        />
      </MemoryRouter>,
      {
        store,
      }
    );
    const { getByText, queryByText } = wrapper;
    expect(getByText("OK")).toBeInTheDocument();
    await act(async () => {
      await fireEvent.click(getByText(/OK/i));
    });
    expect(store.dispatch).toHaveBeenCalledTimes(5);
    expect(hideModal).toHaveBeenCalledTimes(1);
    expect(queryByText("Cancel")).not.toBeInTheDocument();
  });
});
