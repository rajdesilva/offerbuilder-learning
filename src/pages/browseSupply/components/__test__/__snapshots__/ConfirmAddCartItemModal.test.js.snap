// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmAddCartItemModal component test cases create ConfirmAddCartItemModal components snapshot 1`] = `null`;
